src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/alias/TableAliasPrefixRemoveRule.java

```diff
@@ -0,0 +1,109 @@
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.rewriter.alias;
+
+import com.alibaba.druid.sql.ast.expr.SQLIdentifierExpr;
+import com.alibaba.druid.sql.ast.expr.SQLQueryExpr;
+import com.alibaba.druid.sql.ast.statement.SQLExprTableSource;
+import com.alibaba.druid.sql.ast.statement.SQLJoinTableSource;
+import com.alibaba.druid.sql.dialect.mysql.visitor.MySqlASTVisitorAdapter;
+import com.amazon.opendistroforelasticsearch.sql.rewriter.RewriteRule;
+import com.amazon.opendistroforelasticsearch.sql.rewriter.subquery.utils.FindSubQuery;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+/**
+ * Un-alias field name if it's using full table name as prefix.
+ */
+public class TableAliasPrefixRemoveRule implements RewriteRule<SQLQueryExpr> {
+
+    /** Table aliases in FROM clause. Store table name for those without alias. */
+    private final Set<String> tableAliasesToRemove = new HashSet<>();
+
+    @Override
+    public boolean match(SQLQueryExpr root) {
+        if (hasSubQuery(root)) {
+            return false;
+        }
+        collectTableAliasesThatCanBeRemoved(root);
+        return !tableAliasesToRemove.isEmpty();
+    }
+
+    @Override
+    public void rewrite(SQLQueryExpr root) {
+        removeUnAliasedTableNamePrefix(root);
+    }
+
+    private boolean hasSubQuery(SQLQueryExpr root) {
+        FindSubQuery visitor = new FindSubQuery();
+        root.accept(visitor);
+        return visitor.hasSubQuery();
+    }
+
+    private void collectTableAliasesThatCanBeRemoved(SQLQueryExpr root) {
+        visitNonJoinedTable(root, tableExpr -> {
+            Table table = new Table(tableExpr);
+            if (table.hasAlias()) {
+                tableAliasesToRemove.add(table.alias());
+                table.removeAlias();
+            } else {
+                tableAliasesToRemove.add(table.name());
+            }
+        });
+    }
+
+    private void removeUnAliasedTableNamePrefix(SQLQueryExpr root) {
+        visitColumnName(root, idExpr -> {
+            Identifier field = new Identifier(idExpr);
+            if (field.hasPrefix() && tableAliasesToRemove.contains(field.prefix())) {
+                field.removePrefix();
+            }
+        });
+    }
+
+    private void visitNonJoinedTable(SQLQueryExpr root,
+                                     Consumer<SQLExprTableSource> visit) {
+        root.accept(new MySqlASTVisitorAdapter() {
+            @Override
+            public boolean visit(SQLJoinTableSource x) {
+                return false; // Avoid visiting table name in any join clause
```

Comment:
Can we change comment to reflect we also avoid nested query.

Reply:
Will do.